### PR TITLE
Add an API tutorial for Estimators

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is to revive the [notebook](https://github.com/gammapy/gammapy-meetings/blob/master/user-meetings/2022/2022-01-20/estimators.ipynb) from @adonath 

A copy of the notebook can be found at this [gist](https://gist.github.com/Astro-Kirsty/4402da2315bf4e23a42f59c951c30556)

### CODE REVIEW

File: examples/tutorials/api/estimators.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+# for debugging. The `~gammapy.maps.region.ndmap.RegionNDMap` allows for plotting of multidimensional data
+# as well, by specifying the primary `axis_name`:
+
+
+fp_result.counts.plot(axis_name="energy")
+plt.show()
+
+######################################################################
+#
+fp_result.npred.plot(axis_name="energy")
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:183]: Maybe you can remind what is npred, npred_excess... as reminder in this tutorial

File: examples/tutorials/api/estimators.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+map_estimator = TSMapEstimator(
+    model=model,
+    energy_edges=[0.1, 0.3, 1, 3, 10] * u.TeV,
+    n_sigma=1,
+    n_sigma_ul=2,
+    selection_optional=None,
+    n_jobs=8,
+    kernel_width=1 * u.deg,
+    sum_over_energy_groups=True,
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:274]: Does it make sense to describe quickly these arguments, even if it is done in the API documentation? There are not obvious for non-expert

File: examples/tutorials/api/estimators.py [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+######################################################################
+# Estimator configuration
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+model = SkyModel(
+    spectral_model=PowerLawSpectralModel(), spatial_model=PointSpatialModel()
+)
+
+map_estimator = TSMapEstimator(
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:266]: The global section is about flux map estimation... Here, the TSMapEstimator is used, and later the FluxProfileEstimator. Should we present them quickly, summarized their features, before entering into the coding details?

File: examples/tutorials/api/estimators.py [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+
+-  **Hypothesis Testing**: Estimations are based on testing a reference model
+   against a null hypothesis, deriving flux and significance values.
+
+-  **Estimation via Two Methods**:
+
+   -   **Model Fitting (Forward Folding)**: Refit the flux of a model component
+       within specified energy, time, or spatial regions.
+   -   **Excess Calculation (Backward Folding)**: Use the analytical solution by Li and Ma
+       for significance based on excess counts, currently available in `~gammapy.estimators.ExcessMapEstimator`.
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:22]: The ExcessMapEstimator is mentioned here, but never used. Why?

**Review by AtreyeeS — CHANGES_REQUESTED** [review]
Thank you for the revival @Astro-Kirsty 
My main comment is that parts of this notebook are already exposed in our tutorials, so they can be removed. This notebook can focus on explaining how to maniputale/visualise an estimator object

File: examples/tutorials/api/estimators.py [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+lightcurves, flux maps and profiles via a common API.
+
+
+Key Features
+------------
+
+-  **Hypothesis Testing**: Estimations are based on testing a reference model
+   against a null hypothesis, deriving flux and significance values.
+
+-  **Estimation via Two Methods**:
```
[the remark this answers is not in the document]
Comment (Astro-Kirsty) [reply examples/tutorials/api/estimators.py:17]: The purpose was to provide a quick summary here, and refer to the full docs

[the remark this answers is not in the document]
Comment (AtreyeeS) [reply examples/tutorials/api/estimators.py:17]: The wording is exactly the same in the two places. I think the docs page was copied from Axel's notebook :)

File: examples/tutorials/api/estimators.py (hunk 3 above)
Comment (QRemy) [reply examples/tutorials/api/estimators.py]: @AtreyeeS in my opinion if we make an API tutorial about estimators it is worth showing the `TSMapEstimator` again here.

File: examples/tutorials/api/estimators.py [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+)
+
+geom = RegionGeom.create(region=regions)
+ax = counts_image.smooth("0.1 deg").plot()
+geom.plot_region(ax=ax, color="w")
+plt.show()
+
+######################################################################
+#
+flux_profile_estimator = FluxProfileEstimator(
```
[the remark this answers is not in the document]
Comment (QRemy) [reply examples/tutorials/api/estimators.py]: @AtreyeeS in my opinion if we make an API tutorial about estimators it is worth showing the `FluxProfileEstimator ` again here.

File: examples/tutorials/api/estimators.py [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+datasets = Datasets()
+
+path = make_path("$GAMMAPY_DATA/joint-crab/spectra/hess/")
+
+for filename in path.glob("pha_obs*.fits"):
+    dataset = SpectrumDatasetOnOff.read(filename)
+    datasets.append(dataset)
+
+print(datasets)
```
Comment (QRemy) [inline examples/tutorials/api/estimators.py:58]:
```suggestion
```

**Review by QRemy — COMMENTED** [review]
Thanks @Astro-Kirsty, in my opinion if we make an API tutorial about estimators it is worth showing them all even if they are used in other places of the documentation, I wouldn't remove the TSMapEstimator and TSFluxProfileEstimator  as suggested by @AtreyeeS (we can argue that the FluxPointEstimator is also shown in other places and then there is nothing left in this notebook).

In order  to shorten this notebook I would split it in in two : this one that show how to configure and run the estimators and another API tutorial about Flux maps that show how to manipulate an plot the content of the flux maps.
In particular cells 7 to 22 and cells 36-41 would fit better in a second tutorial about Flux maps API.

File: examples/tutorials/api/estimators.py (hunk 3 above)
Comment (AtreyeeS) [reply examples/tutorials/api/estimators.py]: How about adding a link to the other notebook, where it is explained in more details?

Comment (QRemy) [reply examples/tutorials/api/estimators.py]: and here we could just read the results from the other notebooks

File: examples/tutorials/analysis-2d/detect.py [hunk 8]
```diff
@@ -150,18 +150,47 @@
...
+
+######################################################################
+# Here we show a full configuration of the estimator. We reminde the user of the meaning
+# of the various quantities:
+#
+# -  ``model``: a `~SkyModel` which is converted to a source model kernel
+# -  ``kernel_width``: the width for the above kernel
+# -  ``n_sigma``: number of sigma for the flux error
+# -  ``n_sigma_ul``: the number of sigma for the flux upper limits
+# -  ``selection_optional``: what maps to compute
```
Comment (AtreyeeS) [inline examples/tutorials/analysis-2d/detect.py:162]:
```suggestion
# -  ``selection_optional``: what optional maps to compute
```

File: examples/tutorials/analysis-2d/detect.py [hunk 9]
```diff
@@ -179,6 +208,18 @@
 ax3.set_title("Iteration map")
 plt.show()
 
+
+######################################################################
+# The data is represented internally by a reference model and "norm" factors:
```
Comment (AtreyeeS) [inline examples/tutorials/analysis-2d/detect.py:213]:
```suggestion
# The flux in each pixel is obtained by multiplying a reference model with a normalisation factor:
```

File: examples/tutorials/api/estimators.py [hunk 10]
```diff
@@ -0,0 +1,259 @@
+"""
+Estimators
+==========
+
+This tutorial provides an overview of the `Estimator` API. All estimators live in the
+`gammapy.estimators` sub-module offers algorithms and classes for high-level flux and
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:6]:
```suggestion
`gammapy.estimators` sub-module, and offer algorithms and classes for high-level flux and
```

File: examples/tutorials/api/estimators.py [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+significance estimation, through a common functionality such as estimation of flux points,
+lightcurves, flux maps and profiles via a common API.
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:8]: Maybe rephrase this sentence? Its a bit too long

File: examples/tutorials/api/estimators.py [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+######################################################################
+# The `~gammapy.estimators.FluxPointsEstimator` estimates flux points for a given list of datasets,
+# energies and spectral model. Now we prepare the flux point estimation:
+#
+
+energy_edges = np.geomspace(0.7, 100, 9) * u.TeV
+
+fp_estimator = FluxPointsEstimator(
+    source="crab",
+    energy_edges=energy_edges,
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:86]: Maybe show the full configuration here and explain the `reoptimize` feature

**Review by AtreyeeS — COMMENTED** [review]
As discussed in the dev call yesterday, you can read the `PKS 2155 light curve and some flux maps from gammapy data` and show the data structure of such objects.
The main point to emphasise would be that they all follow a very similar structure

**Comment by Astro-Kirsty** [comment]
> As discussed in the dev call yesterday, you can read the `PKS 2155 light curve and some flux maps from gammapy data` and show the data structure of such objects. The main point to emphasise would be that they all follow a very similar structure

Do we still want to introduce the way flux points are actually computed as I do in the tutorial currently? I think it makes sense to show the user how you would produce these in full yourself, and then afterwards import the ones you say and show the common functionality.

File: examples/tutorials/analysis-2d/detect.py [hunk 13]
```diff
@@ -150,18 +150,47 @@
 spectral_model = PowerLawSpectralModel(amplitude="1e-22 cm-2 s-1 keV-1", index=2)
 model = SkyModel(spatial_model=spatial_model, spectral_model=spectral_model)
 
+
+######################################################################
+# Here we show a full configuration of the estimator. We reminde the user of the meaning
```
Comment (AtreyeeS) [inline examples/tutorials/analysis-2d/detect.py:155]:
```suggestion
# Here we show a full configuration of the estimator. We remind the user of the meaning
```

File: examples/tutorials/api/estimators.py [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+Estimators
+==========
+
+This tutorial provides an overview of the `Estimator` API. All estimators live in the
+`gammapy.estimators` sub-module, offering a range of algorithms and classes for high-level flux and
+significance estimation. This is accomplished through a common functionality such as estimation of
+flux points, light curves, flux maps and profiles via a common API.
+
+
+Provides algorithms and classes for flux and significance estimation. Common functionality for flux points,
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:11]: This para can be combined with the previous one

File: examples/tutorials/api/estimators.py [hunk 15]
```diff
@@ -0,0 +1,286 @@
...
+    optimize_opts={"backend": "minuit"},
+    confidence_opts={"backend": "scipy"},
+)
+
+
+######################################################################
+# Define the fully configured flux points estimator:
+#
+
+energy_edges = np.geomspace(0.7, 100, 9) * u.TeV
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:104]: Why geomspace?

Comment (AtreyeeS) [reply examples/tutorials/api/estimators.py:104]: Isn't it better to show how to downsample the original energy axis so that the edges can be perfectly aligned?

File: examples/tutorials/api/estimators.py [hunk 16]
```diff
@@ -0,0 +1,286 @@
...
+# Load the light curve for the PKS 2155-304 as a `~gammapy.estimators.FluxPoints` object.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+lightcurve = FluxPoints.read(
+    "$GAMMAPY_DATA/estimators/pks2155_hess_lc/pks2155_hess_lc.fits", format="lightcurve"
+)
+
+display(lightcurve.available_quantities)
+
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:272]: Did we decide against showing the data structure of the `lightcurve` here?

File: examples/tutorials/api/estimators.py [hunk 17]
```diff
@@ -0,0 +1,286 @@
...
+energy_edges = np.geomspace(0.7, 100, 9) * u.TeV
+
+fp_estimator = FluxPointsEstimator(
+    source="crab",
+    energy_edges=energy_edges,
+    n_sigma=1,
+    n_sigma_ul=2,
+    selection_optional="all",
+    fit=fit,
+    reoptimize=False,
```
Comment (AtreyeeS) [inline examples/tutorials/api/estimators.py:113]: Please show how to configure the norm here. That is often required for upper limit calculations

File: examples/tutorials/api/estimators.py (hunk 16 above)
Comment (Astro-Kirsty) [reply examples/tutorials/api/estimators.py:272]: I believe so, but happy to adjust if you feel it is necessary!

File: examples/tutorials/api/estimators.py (hunk 15 above)
Comment (Astro-Kirsty) [reply examples/tutorials/api/estimators.py:101]: Instead I added to the note to ensure users know how this could be done. 
Given the low statistics at lower energies for this example, it is easier to define an energy binning, otherwise we would need to downsample and then select only the higher energies (this looks quite messy)

File: examples/tutorials/api/estimators.py [hunk 18]
```diff
@@ -0,0 +1,296 @@
...
+fp_result = fp_estimator.run(datasets=datasets)
+
+######################################################################
+# Accessing and visualising the results
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+print(fp_result)
+
+######################################################################
+# We can specify the SED type to plot:
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:153]: I realise that SED is not in the glossary...

File: examples/tutorials/api/estimators.py [hunk 19]
```diff
@@ -0,0 +1,296 @@
...
+#
+fp_result.plot(sed_type="dnde")
+plt.show()
+
+######################################################################
+# From the above we can see that we access to many quantities. We can also access
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:159]:
```suggestion
# From the above, we can see that we access to many quantities. We can also access
```

Comment (bkhelifi) [reply examples/tutorials/api/estimators.py:159]: I would have put this line just before l185...

File: examples/tutorials/api/estimators.py [hunk 20]
```diff
@@ -0,0 +1,296 @@
...
+######################################################################
+# Dataset specific quantities ("counts like")
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# While the flux estimate and associated errors are common to all datasets,
+# the result also stores some dataset specific quantities, which can be useful
+# for debugging.
+# Here we remind the user of the meaning of the forthcoming quantities:
+#
+# -  ``counts``: predicted counts from the null hypothesis.
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:226]:
```suggestion
# -  ``counts``: predicted counts from the null hypothesis,
```

File: examples/tutorials/api/estimators.py [hunk 21]
```diff
@@ -0,0 +1,296 @@
...
+# -  ``npred``: predicted number of counts from best fit hypothesis.
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:227]:
```suggestion
# -  ``npred``: predicted number of counts from best fit hypothesis,
```

File: examples/tutorials/api/estimators.py [hunk 22]
```diff
@@ -0,0 +1,296 @@
...
+# -  ``npred_excess``: predicted number of excess counts from best fit hypothesis.
+#
```
Comment (bkhelifi) [inline examples/tutorials/api/estimators.py:229]: Should we also remind the same for the background?

**Review by bkhelifi — COMMENTED** [review]
Thanks a lot @Astro-Kirsty . Here are my last comments...

**Review by bkhelifi — DISMISSED** [review]
Thanks a lot ;)

**Comment by bkhelifi** [comment]
I forget to ask whether one should waiting for the merging of #5454 to make a proper link to the glossary...

**Review by AtreyeeS — DISMISSED** [review]
Thanks @Astro-Kirsty 
Since we now store the likelihood scans in each bin, I think it will be very useful if you can show one example.

**Comment by QRemy** [comment]
Resolves #3716
